_adr/88/index.adoc

```diff
@@ -1,48 +1,70 @@
 ---
 num: 88
-title: Authorization for control plane admin APIs
-status: "Draft" # One of Draft, Accepted, Rejected
+title: Authorization for fleet-manager admin APIs
+status: "Draft"
 authors:
-  - "akoserwal"
+  - "ppaszki"
 tags:
-  - "sso" # e.g. kafka, connectors, registry
+  - "sso"
 applies_padrs: # What PADRs does this ADR apply?
 applies_patterns: # What APs does this ADR apply?
 ---
 
-// Top style tips:
-// * Use one sentence per line
-// * No unexpanded acronyms
-// * No undefined jargon
-
-// No need for a title heading, it's added by the template
+:rover-groups: https://rover.redhat.com/groups
 
 ## Context and Problem Statement
-// What is the background against which this decision is being taken?
+The purpose of this ADR is to define a standardized approach for securing admin API in a fleet manager
 
 ## Goals
-// Bulleted list of outcomes that this ADR, if accepted, should help achieve
+* Provide standardized way to secure admin API endpoints for fleet managers
 
 ## Non-goals
-// Bulleted list of outcomes that this ADR is not trying to achieve.
 
 ## Current situation
-// Where are we now?
 
 ## Proposal
-// What is the decision being proposed
+
+image::adr-88-admin-api-architecture-diagram.png[architecture diagram]
+
+A Keycloak realm should be used to authorize access to admin API endpoint(s). 
+Such realm should contain roles that allow access to endpoints to the privileged API users.
+Recommended approach would be to have set of roles per environment and name them in the way that indicates the permissions they grant as follows:
+
+* <unique-fleet-manager-prefix>-admin-read-stage
+* <unique-fleet-manager-prefix>-admin-write-stage
+* <unique-fleet-manager-prefix>-admin-full-stage
+
+* <unique-fleet-manager-prefix>-admin-read-prod
+* <unique-fleet-manager-prefix>-admin-write-prod
+* <unique-fleet-manager-prefix>-admin-full-prod
+
+These roles map to the following permissions for the environment included as the role postfix (e.g. `prod`):
+
+* \*-read-* - list permissions
+* \*-write-* - list/ update permissions
+* \*-full-* - full permissions (list/ update/ delete)
+
+Once Keycloak instance is set up, a privileged user uses their credentials to get a JWT token from Keycloak.
+The JWT token will be sent to a admin API in a HTTP request.
+Admin API authorization middleware verifies token signature with Public JWKS Certificate and roles included in the JWT token are checked against roles specified in the fleet manager.
+Matching role for given admin endpoint will grant access to the action associated with that endpoint.
+Failure to match the required role will result in restricting access to the endpoint.
+
+image::adr-88-authentication-flow.png[architecture diagram]
+
+## Suggested implementation for Red Hat Internal Keycloak
```

Comment:
hum @tombentley , how can we deal with such situation? (internal RHT sauce descriptions? The idea was to use the same model as SOP layering. But i can see it might not be ready.

1. We accept the imperfection
2. we use the old ADR folder model as a addendum `ADR-88 Authorization for fleet-manager admin APIs -- Red Hat Addendum` ina  doc and point to this one + add the red hat specific paragraph below?
3. other

Reply:
It's a great question. In theory 1 should be fine, because `security != obscurity`. But I think 2 would be pragmatic all the same.

Reply:
@emmanuelbernard @tombentley - I have updated this section and created a document. please let me know if that's what you meant. I have also addressed all other comments, so I think this ADR is ready for a final review